.github/actions/create-bwc-build/action.yaml

```diff
@@ -0,0 +1,47 @@
+name: 'Create a backwards compatible ready build'
+description: 'Checkouts the official version of a the Security plugin and builds it so it can be used for BWC tests'
+
+inputs:
+  plugin-branch:
+    description: 'The branch of the plugin that should be built, e.g "2.2", "1.x"'
+    required: true
+  
+outputs:
+  built-version:
+    description: 'The version of OpenSearch that was associated with this branch'
+    value: ${{ steps.get-opensearch-version.outputs.built-version }}
+
+runs:
+  using: "composite"
+  steps:
+      # Configure longpath names if on Windows
+    - name: Enable Longpaths if on Windows
+      if: ${{ runner.os == 'Windows' }}
+      run: git config --system core.longpaths true
+      shell: pwsh
+
+    - uses: actions/checkout@v3
+      with:
+        repository: opensearch-project/security
+        ref: ${{ inputs.plugin-branch }}
+        path: ${{ inputs.plugin-branch }}
+
+    - name: Build
+      uses: gradle/gradle-build-action@v2
+      with:
+        arguments: assemble -Dbuild.snapshot=false
+        build-root-directory: ${{ inputs.plugin-branch }}
+
+    # This might not work in windows
+    - id: get-opensearch-version
+      run: |
+        export extractedVersion=$(./gradlew properties -Dbuild.snapshot=false | grep ^version: | awk '{split($0, a, ": ");print a[2]}')
```

Comment:
While I love this, I have a suspicion Windows will not handle the `awk` properly. Could be wrong though.

Reply:
Bash on windows is amazing! See the git hub action passes at this part, command executes [1] then see the value is used [2] to create the directory variable.

I also tested on my windows desktop in the Git Bash
```
petern@SEA-1800605833 MINGW64 ~/Documents/GitHub/security (main)
$ export extractedVersion=$(./gradlew properties -Dbuild.snapshot=false | grep ^version: | awk '{split($0, a, ": ");print a[2]}')
Picked up JAVA_TOOL_OPTIONS: -Dlog4j2.formatMsgNoLookups=true

petern@SEA-1800605833 MINGW64 ~/Documents/GitHub/security (main)
$ echo $extractedVersion
3.0.0.0
```

Unfortunately the tests are still broken because we don't have 2.5.0 version :/


[1]  https://github.com/opensearch-project/security/actions/runs/3463162053/jobs/5784787988#step:4:590
[2] https://github.com/opensearch-project/security/actions/runs/3463162053/jobs/5784787988#step:4:603

Reply:
Well I stand corrected then! I think my use of Powershell must have messed with a similar implementation I had made a while back then. Regardless this seems great then. Maybe another opportunity for you to make one of your little sub actions if you wanted even. I bet a lot of people/teams would be interested in it :)

Reply:
Darn your reasonable request https://github.com/peternied/get-opensearch-version!

Reply:
Lovely